### DISCUSSION TITLE
How does the regexManager work

### DISCUSSION THREAD

**Discussion by anwbtom** [renovatebot/renovate#20549] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.146.1

### If you're self-hosting Renovate, select which platform you are using.

None

### If you're self-hosting Renovate, tell us what version of the platform you run.

Gitlab EE

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We're trying to make sure that renovatebot can update the versions of helm installs of our helm_releases.  

During our tests it seems that it can never be fully parsed, but that it might be resolved with regexMatchers, debug logging of renovate show something like this structure during the run:
```
{
  "packageFile": "cilium.tf",
  "deps": [
    {
      "currentValue": "${var.cilium_chart_version}",
      "depType": "helm_release",
      "depName": "cilium",
      "datasource": "helm",
      "registryUrls": ["https://helm.cilium.io/"],
      "depIndex": 0,
      "updates": [],
      "warnings": [],
      "versioning": "helm",
      "skipReason": "invalid-value"
    }
  ]
},
```
Currently we have a terraform repository with the following structure

```
cilium.tf
other_application.tf
versions.tf
versions.auto.tfvars
```

Example contents of these files:  

`cilium.tf`:  
```
resource "helm_release" "cilium" {
  name             = "cilium"
  repository       = "https://helm.cilium.io/"
  chart            = "cilium"
  version          = var.cilium_chart_version
  namespace        = "cilium"
  timeout          = 300
  force_update     = true
  wait             = true
  wait_for_jobs    = true
  atomic           = true
  
  values           = templatefile("${module.path}/tps/cilium.jinja2")
}
```
`other_application.tf`:  
```
resource "helm_release" "other_application" {
  name             = "other_application"
  repository       = "https://my-awesome-repo.com"
  chart            = "other_application"
  version          = var.other_application_chart_version
  namespace        = "other_application"
  timeout          = 300
  force_update     = true
  wait             = true
  wait_for_jobs    = true
  atomic           = true
  
  values           = templatefile("${module.path}/tps/other_application.jinja2")
}
```
`versions.tf`:
```
variable "cilium_chart_version" {
  description = "Version of Cilium to install"
  type        = string
}

variable "other_application_chart_version" {
  description = "Version of other_application to install"
  type        = string
}
```
`versions.auto.tfvars`: 
```
cilium_chart_version                       = "1.12.6" # https://artifacthub.io/packages/helm/cilium/cilium
other_application_chart_version            = "0.0.1" # https://artifacthub.io/packages/helm/other_applcation/application
```

### What you tried so far.

We tried to write our own regex-matcher and also looked into (to us, seemingly related discussions/issues) like: https://github.com/renovatebot/renovate/discussions/16052).  

We're starting to wondering if this can even work, and if so what would be the best approach?

**Comment by viceice** [comment, marked as answer]
I'm using this regex manager for my terraform variables 😉

https://github.com/visualon/renovate-config/blob/5ad23a52ab2b986a09b1079f52261c59e5ce1e8e/shared.json#L150-L158

**Reply by anwbtom** [reply]
Thank you so much for your answer!  
But this doesn't seem to really help me, does you terraform also follow the structure of setting the values of said variables somewhere else?  
I'm wondering if renovate can regexMatchers can be used when it needs multiple file sources to get to the conclusion of a single "reource" eg. the chart is defined in a separated file, the variable definition is and the value assignement of said variable definition is as well.

The documentation about how it collects stuff and what is important to match on, is quite scarce for me, since I'm quite new at the whole renovate bot thing.

With the starting regexMatcher you've provided I atleast saw that it wanted to replace something, but alas not with the value of the actual variable, just the variable as a different string.  

So am sort of grabbing at straws here.

**Reply by viceice** [reply]
no, regex manager can only work on a single file, that's why I'm using comments for letting renovate know about the dependency.

maybe some time in future renovate can follow variables in terraform manager like in Gradle and Maven.

**Reply by anwbtom** [reply]
Alright, thank you very much for the information and for your effort!
